experts-api/lib/elements-client.js

```diff
@@ -65,12 +74,39 @@ export class Impersonator {
     delete options.abort_controller;
     const id = setTimeout(() => controller.abort(), timeout);
 
-    const response = await this.fetch(url, {
-      ...options,
-      signal: controller.signal
-    });
-    clearTimeout(id);
-    return response;
+    let resp;
+    try {
+      resp = await this.fetch(url, {
```

Comment:
# Example 3 #
This is error is more complicated, in this case we *are* making calls to an external site.  We want the user/dev to know that it's that server that is the problem.  In this case we propagate both gateway timeout errors, and and non-timeout errors.  Here, sympletic has it's own JSON based response body.  We convert this to text and pass along, for more information.